manager/src/main/java/io/aklivity/zilla/manager/internal/commands/install/ZpmInstall.java

```diff
@@ -548,8 +548,7 @@ private void generateDelegate(
                                 assert servicePath.getNameCount() == 1;
                                 String serviceName = servicePath.toString();
                                 String serviceImpl = new String(input.readAllBytes(), UTF_8);
-                                String existing = services.getOrDefault(serviceName, "");
-                                services.put(serviceName, existing.concat("\n").concat(serviceImpl));
+                                services.computeIfAbsent(serviceName, s -> new ArrayList<>()).add(serviceImpl);
```

Comment:
The service implementation can be multi-valued separated by newlines, so we would need to split it here to handle that case properly.